src/file_storage.cpp

```diff
@@ -790,52 +790,52 @@ namespace libtorrent
 		return ret;
 	}
 
-	string_view file_storage::file_name(int index) const
+	string_view file_storage::file_name(file_index_t const index) const
 	{
-		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
+		TORRENT_ASSERT_PRECOND(index < end_file());
 		internal_file_entry const& fe = m_files[index];
 		return fe.filename();
 	}
 
-	std::int64_t file_storage::file_size(int index) const
+	std::int64_t file_storage::file_size(file_index_t const index) const
 	{
-		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
+		TORRENT_ASSERT_PRECOND(index < end_file());
 		return m_files[index].size;
 	}
 
-	bool file_storage::pad_file_at(int index) const
+	bool file_storage::pad_file_at(file_index_t const index) const
 	{
-		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
+		TORRENT_ASSERT_PRECOND(index < end_file());
 		return m_files[index].pad_file;
 	}
 
-	std::int64_t file_storage::file_offset(int index) const
+	std::int64_t file_storage::file_offset(file_index_t const index) const
 	{
-		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
+		TORRENT_ASSERT_PRECOND(index >= file_index_t(0) && index < end_file());
```

Comment:
although, in the future I envision the check to be in the constructor of ``file_index_t``, to make sure it can never be negative